### DISCUSSION TITLE
Weighting indices by other models

### DISCUSSION THREAD

**Discussion by Nathan-Hebert** [sdmTMB/sdmTMB#349] [Q&A] [answered]
I want to weight an index from model A with density predictions from model B. Is there any way to propogate the uncertainty in model B's predictions (and thus weighting) into the uncertainty of the model A index? Thanks!

**Comment by ericward-noaa** [comment, marked as answer]
I don't know that I'm totally following this, but I think yes. I think you could make predictions from model B with `se_fit` = TRUE (or use the `nsim` argument). The SEs could then be used to generate inverse variance weights. These inverse variance weights can then be multiplied by the area of cells for the prediction grid and passed into `get_index()` for model A.

**Reply by Nathan-Hebert** [reply]
That makes sense, thanks!

**Comment by seananderson** [comment]
Using inverse variance weights will achieve the goal of relative weighting based on uncertainty, which might be what you're looking for, but won't propagate the full parameter uncertainty, right?

To fully propagate the uncertainty, your options would be to 

1. Fit both simultaneously (which might be possible with tinyVAST, depending on what you're looking to do; [body condition example here](https://vast-lib.github.io/tinyVAST/articles/web_only/condition.html)) or write a bespoke model yourself, [say with RTMB](https://github.com/seananderson/RTMB-TESA-spatial).

2. Take draws from the first model and combine those with draws from the 2nd model to calculate your index (assumes they are independent). This could be with MCMC samples or (more quickly but approximated) using `nsim` to take draws from the joint precision matrix. Although slightly different, Philina English took that approach [in this paper](https://onlinelibrary.wiley.com/doi/abs/10.1002/pan3.10554)—take MCMC samples from two SDMs and then look at expected ratios of the two using those samples.

**Reply by Nathan-Hebert** [reply]
Thank you, this is very helpful.
